src/Sarif.Multitool/Rules/SARIF1021.MissingToolVersion.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class MissingToolVersion : SarifValidationSkimmerBase
+    {
+        private readonly MultiformatMessageString _fullDescription = new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF1021_MissingToolVersion
+        };
+
+        public override MultiformatMessageString FullDescription => _fullDescription;
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        /// <summary>
+        /// SARIF1021
+        /// </summary>
+        public override string Id => RuleId.MissingToolVersion;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF1021_Default)
+        };
+
+        protected override void Analyze(Tool tool, string pointer)
+        {
+            // 'Driver' is a required property, hence we do not need a null check for it. 
+            if (string.IsNullOrWhiteSpace(tool.Driver.Version))
+            {
+                LogResult(pointer, nameof(RuleResources.SARIF1021_MissingToolVersion), tool.Driver.Name);
```

Comment:
>pointer [](start = 26, length = 7)

This should be `pointer.AtProperty(SarifPropertyNames.Driver)`. The place to report the error is (for example) `/runs/0/tool/driver` -- that's the object that we're complaining doesn't have the `version` property.

Reply:
OOH, thats wat it does. this explains why i was seeing the line numbers point to parent object in the output. fixing!

---
In reply to: [434787246](https://github.com/microsoft/sarif-sdk/pull/1899#discussion_r434787246) [](ancestors = 434787246)